c_src/zstd_nif.c

```diff
@@ -37,12 +37,17 @@ static ERL_NIF_TERM zstd_nif_decompress(ErlNifEnv* env, int argc, const ERL_NIF_
   if(!enif_inspect_binary(env, argv[0], &bin))
     return enif_make_badarg(env);
 
-  uncompressed_size = ZSTD_getDecompressedSize(bin.data, bin.size);
+  uncompressed_size = ZSTD_getFrameContentSize(bin.data, bin.size);
+
+  if (uncompressed_size==ZSTD_CONTENTSIZE_ERROR) {
+      return enif_make_badarg(env);
+  } else if (uncompressed_size==ZSTD_CONTENTSIZE_UNKNOWN) {
+      uncompressed_size = 10000000;
```

Comment:
It was the first arbitrarily large number I thought of being an approximation of 10MB. I was thinking a single record in a stream will be compressed and the compressed size checked, so we are looking at just under a MB due to calculating the overhead of kinesis record wrapping done in ingest. So we should handle something that decompresses to < 10 times it's original size. If it gets larger than that we would fail and record that data for later inspection.